packages/components/src/components/date-picker/date-picker.tsx

```diff
@@ -1,25 +1,63 @@
-import PropTypes from 'prop-types';
-import React from 'react';
+import React, { type ReactElement } from 'react';
 import { addDays, daysFromTodayTo, toMoment, convertDateFormat, getPosition, isMobile } from '@deriv/shared';
 
-import Input from './date-picker-input.jsx';
-import Calendar from './date-picker-calendar.jsx';
-import Native from './date-picker-native.jsx';
+import Input from './date-picker-input';
+import Calendar from './date-picker-calendar';
+import Native from './date-picker-native';
 import MobileWrapper from '../mobile-wrapper';
 import DesktopWrapper from '../desktop-wrapper';
 import { useOnClickOutside } from '../../hooks/use-onclickoutside';
 
-const DatePicker = React.memo(props => {
+type TDatePicker = {
+    error_messages: string[];
+    label: string;
+    is_alignment_top: boolean;
+    date_format: string;
+    disabled: boolean;
+    mode: string;
+    max_date: moment.Moment;
+    min_date: moment.Moment;
+    start_date: moment.Moment;
+    name: string;
+    onFocus: () => void;
+    portal_id: string;
+    placeholder: string;
+    required: boolean;
+    type: string;
+    value: moment.Moment | string;
+    data_testid: string;
+    display_format: string;
+    error: string;
+    footer: ReactElement;
+    id: string;
+    has_range_selection: boolean;
+    onBlur: () => void;
+    onChange: ({
+        date,
+        duration,
+        target,
+    }: {
+        date?: string;
+        duration?: number | null | string;
+        target?: { name: string; value: number | string | moment.Moment | null };
+    }) => void;
+    alignment: string;
+    keep_open: boolean;
+};
```

Comment:
```suggestion
type TDatePickerOnChangeEvent = {
    date?: string;
    duration?: number | null | string;
    target?: { name: string; value: number | string | moment.Moment | null };
};

type TDatePicker = Omit<TDatePickerNativeProps & TDatePickerInputProps & TDatePickerCalendarProps, 'value'> &  {
    mode: string;
    start_date: moment.Moment;
    value: moment.Moment | string;
    onChange: (e: TDatePickerOnChangeEvent) => void;
};
```
We can extend children's props types here to avoid duplication of the same fields🙂
Also, please kindly check if we are using `is_alignment_top` or `error_messages` anywhere. I removed them from my suggestion because it seems like `error_messages` are no longer used in packages/components/src/components/date-picker/date-picker-input.jsx, and `is_alignment_top` is not used in any of the children either.